recipes/ulfius/all/conanfile.py

```diff
@@ -0,0 +1,204 @@
+            self.options.enable_websockets = False
+            del self.options.fPIC
+
+    def validate(self):
+        if self.options.with_gnutls:
+            raise ConanInvalidConfiguration("with_gnutls=True is not yet implemented due to mossing gnutls CCI recipe")
+        if self.settings.os == "Windows" and self.options.enable_websockets:
+            raise ConanInvalidConfiguration("ulfius does not support with_websockets=True is not supported on Windows")
+
+    def configure(self):
+        if self.options.shared:
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+        try:
+            del self.settings.compiler.libcxx
+        except Exception:
+            pass
+        try:
+            del self.settings.compiler.cppstd
+        except Exception:
+            pass
+
+    def requirements(self):
+        self.requires("orcania/2.3.1")
+        self.requires("libmicrohttpd/0.9.75")
+        if self.options.with_yder:
+            self.requires("yder/1.4.18")
+        if self.options.with_jansson:
+            self.requires("jansson/2.14")
+        if self.options.with_libcurl:
+            self.requires("libcurl/7.85.0")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["BUILD_SHARED"] = self.options.shared
+        tc.variables["BUILD_STATIC"] = not self.options.shared
+        tc.variables["DOWNLOAD_DEPENDENCIES"] = False
+        tc.variables["WITH_GNUTLS"] = self.options.with_gnutls
+        tc.variables["WITH_WEBSOCKETS"] = self.options.enable_websockets
+        tc.variables["WITH_CURL"] = self.options.with_libcurl
+        tc.variables["WITH_JANSSON"] = self.options.with_jansson
+        tc.generate()
+
+        generator_dir = os.path.join(self.build_folder, "generators")
+        self.dependencies["libmicrohttpd"].cpp_info.set_property("cmake_file_name", "MHD")
+        self.dependencies["libmicrohttpd"].cpp_info.set_property("cmake_target_name", "MHD::MHD")
+        self.dependencies["libmicrohttpd"].cpp_info.set_property("cmake_module_file_name", "MHD")
+        self.dependencies["libmicrohttpd"].cpp_info.set_property("cmake_module_target_name", "MHD::MHD")
+        mhd_version_cmake = os.path.join(generator_dir, "mhd_version.cmake")
+        self.dependencies["libmicrohttpd"].cpp_info.set_property("cmake_build_modules", [mhd_version_cmake])
+        save(self, mhd_version_cmake, textwrap.dedent(f"""
+            set(MHD_VERSION_STRING {self.dependencies['libmicrohttpd'].ref.version})
+        """))
+
+        if self.options.shared:
+            self.dependencies["orcania"].cpp_info.set_property("cmake_target_name", "Orcania::Orcania")
```

Comment:
`self.dependencies` is read-only. https://docs.conan.io/en/latest/reference/conanfile/dependencies.html?highlight=read%20only#dependencies-interface

Reply:
I was asking about this in https://github.com/conan-io/conan/issues/12357.
Making it locally modifiable could avoid a lot of patches.